Add TextStream operator for FocusableElementSearchResult

aprotyas opened the pull request:
#### edb317a5fc3cd04b68b854e44c364441686fb193
<pre>
Add TextStream operator for FocusableElementSearchResult
<a href="https://bugs.webkit.org/show_bug.cgi?id=300220">https://bugs.webkit.org/show_bug.cgi?id=300220</a>

Reviewed by Aditya Keerthi.

Several focus traversal methods in FocusController use this return type,
so a TextStream operator helps with debug logging.

No functional change, so no new tests.

* Source/WebCore/page/FocusController.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/page/FocusControllerTypes.h:

Canonical link: <a href="https://commits.webkit.org/301072@main">https://commits.webkit.org/301072@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/02bf3d5e8f3c580ff18656a6476949226e18a897

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124873 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44544 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35280 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131719 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76773 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7baea1c4-337d-45c5-bb51-362836640fd5) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126750 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45245 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53112 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95028 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/63051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c0ee8163-7706-432a-9da8-c27903cfd832) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127827 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36099 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111677 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75585 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/6cd33321-84f1-4e26-ba70-2c8b31cef1a5) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35036 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/29834 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75196 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105859 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30062 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134387 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51714 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39515 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103505 "30 flakes 76 failures") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52131 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107892 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103268 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48628 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26916 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48735 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/19568 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51590 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/57398 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50979 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54335 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52672 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b94adf53511167c8f153fc2a410f6b0067b059ae. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51853

aprotyas commented on Source/WebCore/page/FocusController.cpp:
```diff
@@ -1382,4 +1383,14 @@ Seconds FocusController::timeSinceFocusWasSet() const
+    auto elementDescription = element ? element->debugDescription() : "null"_s;
+    bool didContinueSearchInRemoteFrame = continuedSearchInRemoteFrame == ContinuedSearchInRemoteFrame::Yes;
+    bool didRelinquishFocusToChrome = relinquishedFocusToChrome == RelinquishedFocusToChrome::Yes;
+    ts << "element: " << elementDescription << ", continued search in remote frame: " << didContinueSearchInRemoteFrame << ", relinquished focus to chrome: " << didRelinquishFocusToChrome;
```
Fixed in https://github.com/WebKit/WebKit/commit/02bf3d5e8f3c580ff18656a6476949226e18a897

aprotyas committed:
Add TextStream operator for FocusableElementSearchResult
https://bugs.webkit.org/show_bug.cgi?id=300220

Reviewed by Aditya Keerthi.

Several focus traversal methods in FocusController use this return type,
so a TextStream operator helps with debug logging.

No functional change, so no new tests.

* Source/WebCore/page/FocusController.cpp:
(WebCore::operator<<):
* Source/WebCore/page/FocusControllerTypes.h:

Canonical link: https://commits.webkit.org/301072@main
